src/connection.ts

```diff
@@ -655,6 +665,20 @@ export class Connection extends EventEmitter {
       }
     }
 
+    // Allow consumer to choose to fulfill each packet and/or perform other logic before fulfilling
+    if (this.shouldFulfill && incomingAmount.greaterThan(0)) {
+      await this.shouldFulfill(this.connectionId, requestPacket.sequence, incomingAmount).catch(async err => {
```

Comment:
This should expose the connection tag to the shouldFulfill function. Technically they could parse it out of connectionId but we should abstract the actual formatting of that away.